cpu/cortexm_common/thread_arch.c

```diff
@@ -311,23 +320,15 @@ void __attribute__((naked)) __attribute__((used)) isr_pendsv(void) {
     "stmdb  r0!,{r4-r11}              \n" /* save regs */
     "stmdb  r0!,{lr}                  \n" /* exception return value */
 #endif
-    "ldr    r1, =sched_active_thread  \n" /* load address of current tcb */
-    "ldr    r1, [r1]                  \n" /* dereference pdc */
-    "str    r0, [r1]                  \n" /* write r0 to pdc->sp */
-    "bl     isr_svc                   \n" /* continue with svc */
-    );
-}
+    "str    r0, [r1]                  \n" /* write r0 to thread->sp */
+
+    /* current thread context is now saved */
 
-void __attribute__((naked)) __attribute__((used)) isr_svc(void) {
-    __asm__ volatile (
-    /* SVC handler entry point */
-    /* PendSV will continue here as well (via jump) */
-    ".thumb_func                      \n"
-    /* perform scheduling */
-    "bl     sched_run                 \n"
-    /* restore context and return from exception */
-    ".thumb_func                      \n"
     "context_restore:                 \n"
+
+    "bl     sched_run                 \n" /* perform scheduling */
```

Comment:
`sched_run` should not be part of `context_restore` IMO, I understand this is because of the branch at L294, but to keep the same ordering I would rather have:

```suggestion
    "run_shed:                         \n"
    "bl     sched_run                 \n" /* perform scheduling */
    "context_restore:                 \n"
```

Reply:
hm, those are goto labels. Noone jumps to "run_shed", so we on't need it. Maybe "context_restore" could be named "context_select_and_restore" or something.

Reply:
> hm, those are goto labels. Noone jumps to "run_shed", so we on't need it. Maybe "context_restore" could be named "context_select_and_restore" or something.

I would branch to it on L294 `"beq run_sched             \n" /*     goto context_restore   */`

I just don't like `sched_run` being part of `context_restore`. I would also be Ok with a name change.

Reply:
renamed it to "select_and_restore_context"

Reply:
ok